app/src/main/java/org/kiwix/kiwixmobile/localFileTransfer/WifiDirectManager.kt

```diff
@@ -243,26 +238,39 @@ class WifiDirectManager @Inject constructor(
     if (BuildConfig.DEBUG) {
       Log.d(TAG, "Starting handshake")
     }
-    peerGroupHandshakeAsyncTask = PeerGroupHandshakeAsyncTask(this)
-      .also { it.execute() }
+    var inetAddress: InetAddress?
+    lifecycleCoroutineScope.launch {
+      inetAddress = if (isFileSender) {
+        SenderHandShake(this@WifiDirectManager).handshake()
```

Comment:
to avoid having `.handshake()` declared twice in this file store sender or receiver in a variable as per my example